l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -346,34 +353,52 @@ library Decoder {
         (vars.unencryptedLogsHashKernel2, offsets.unencryptedLogsOffset) =
           computeKernelLogsHash(offsets.unencryptedLogsOffset, _l2Block);
 
-        uint256 commitmentsPerBase = 2 * Constants.COMMITMENTS_PER_TX;
-        uint256 nullifiersPerBase = 2 * Constants.NULLIFIERS_PER_TX;
-
         assembly {
           let baseLeaf := mload(add(vars, 0x40)) // Load the pointer to `vars.baseLeaf`
           let dstPtr := add(baseLeaf, 0x20) // Current position withing `baseLeaf` to write to
+          let leafDataLengthPerBase := mload(constantsLengthsPerBase)
```

Comment:
Think you should rename this as well, it is the `memoryConstantPtr` that you are then looping over to progress nicely.

Reply:
changed it to something much better ahha ;)